app/build.gradle

```diff
@@ -1,306 +0,0 @@
- *
- * Poet Assistant is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * Poet Assistant is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with Poet Assistant.  If not, see <http://www.gnu.org/licenses/>.
- */
-buildscript {
-    repositories {
-        jcenter()
-    }
-    dependencies {
-        classpath 'com.github.ben-manes:gradle-versions-plugin:0.52.0'
-    }
-}
-
-plugins {
-    id 'jacoco'
-    id 'com.google.devtools.ksp'
-}
-apply plugin: 'com.android.application'
-apply plugin: 'com.github.ben-manes.versions'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-kapt'
-android {
-    compileSdkVersion 35
-
-    compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
-    }
-    kotlinOptions {
-        jvmTarget = '1.8'
-    }
-
-    buildFeatures {
-        dataBinding = true
-        buildConfig = true
-    }
-    jacoco {
-        version "0.8.12"
-    }
-    lintOptions {
-        abortOnError true
-        textReport true
-        ignoreWarnings true
-        disable 'RestrictedApi'  // https://stackoverflow.com/questions/45648530/restricted-api-lint-error-when-deleting-table-room-persistence
-        checkReleaseBuilds false
-    }
-
-    defaultConfig {
-        applicationId "ca.rmen.android.poetassistant"
-        namespace "ca.rmen.android.poetassistant"
-        minSdkVersion 21
-        targetSdkVersion 35
-        versionCode 113101
-        versionName "1.31.1"
-        // setting vectorDrawables.useSupportLibrary = true means pngs won't be generated at
-        // build time: http://android-developers.blogspot.fr/2016/02/android-support-library-232.html
-        vectorDrawables.useSupportLibrary = true
-        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
-
-        // The following argument makes the Android Test Orchestrator run its
-        // "pm clear" command after each test invocation. This command ensures
-        // that the app's state is completely cleared between tests.
-        // https://developer.android.com/training/testing/instrumented-tests/androidx-test-libraries/runner#use-android
-        // testInstrumentationRunnerArguments clearPackageData: 'true', coverage: 'true', coverageFilePath: '/data/data/ca.rmen.android.poetassistant.test/'
-
-
-        // used by Room, to test migrations
-        sourceSets {
-            main.java.srcDirs += "$projectDir/src/main/kotlin"
-            main.assets.srcDirs += "${project.buildDir}/generated/license_assets"
```

Comment:
`project.buildDir` as a string has been replaced with `project.layout.buildDirectory` which is a directory object.